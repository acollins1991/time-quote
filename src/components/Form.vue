<template>
<div>
  <form v-on:guessUpdated="updateGuesses">
    Optimistic:
    <br>
    <TimeInput v-bind:guess="'optimisticGuess'" />
    <br>
    Best Guess:
    <br>
    <TimeInput v-bind:guess="'bestGuess'" />
    <br>
    Pessimistic:
    <br>
    <TimeInput v-bind:guess="'pessitimisticGuess'" />
    <br>
    <input type="submit" class="tq-submit bg-blue hover:bg-blue-dark text-white font-bold py-2 px-4 rounded" value="Submit" v-on:click.prevent="emitGuessesSubmitted">
  </form>
</div>
</template>

<script>
import TimeInput from "@/components/TimeInput.vue"

export default {
  name: "Form",
  data(){
    return {
      optimisticGuess: 0,
      bestGuess: 0,
      pessitimisticGuess: 0
    }
  },
  components: {
    TimeInput
  },
  methods: {
    emitGuessesSubmitted: function(){
      const bestGuess = this.bestGuess
      const optimisticGuess = this.optimisticGuess
      const pessitimisticGuess = this.pessitimisticGuess

      this.$emit('guessesSubmitted', { bestGuess, optimisticGuess, pessitimisticGuess })
    }
  }
}
</script>

