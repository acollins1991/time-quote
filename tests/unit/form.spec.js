import { shallowMount, createWrapper } from '@vue/test-utils'
import Form from '@/components/Form.vue'

describe('Form.vue', () => {
  it('renders the form', () => {
    const wrapper = shallowMount(Form)
    expect(wrapper.find('form').exists()).toBe(true)
  })
  it('emits custom estimation event on submission from root', () => {
    const wrapper = shallowMount(Form)
    const submitButton = wrapper.find('input.tq-submit')
    const rootWrapper = createWrapper(wrapper.vm.$root)

    wrapper.vm.$data.bestGuess = 5
    wrapper.vm.$data.optimisticGuess = 5
    wrapper.vm.$data.pessimisticGuess = 5

    submitButton.trigger('click')

    console.log(rootWrapper.emitted().guessesSubmitted[0][0])

    expect(rootWrapper.emitted().guessesSubmitted).toBeTruthy()
    expect(rootWrapper.emitted().guessesSubmitted[0][0].bestGuess).toEqual(5)
    expect(rootWrapper.emitted().guessesSubmitted[0][0].optimisticGuess).toEqual(5)
    expect(rootWrapper.emitted().guessesSubmitted[0][0].pessimisticGuess).toEqual(5)

  })
})
