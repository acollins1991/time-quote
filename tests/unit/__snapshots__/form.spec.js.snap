// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form.vue renders correctly 1`] = `
<div>
  <form>
    <span
      class="block px-3 mt-2 mb-1 font-sans text-2xl"
    >
      Optimistic:
    </span>
     
    <timeinput-stub
      guess="optimisticGuess"
    />
     
    <span
      class="block px-3 mt-2 mb-1 font-sans text-2xl"
    >
      Best Guess:
    </span>
     
    <timeinput-stub
      guess="bestGuess"
    />
     
    <span
      class="block px-3 mt-2 mb-1 font-sans text-2xl"
    >
      Pessimistic:
    </span>
     
    <timeinput-stub
      guess="pessimisticGuess"
    />
     
    <div
      class="px-3 mt-2 mb-3"
    >
      <input
        class="tq-submit bg-blue hover:bg-blue-dark text-white font-bold py-2 px-4 rounded"
        type="submit"
        value="Submit"
      />
    </div>
  </form>
</div>
`;
